### DISCUSSION TITLE
Is there any example for using SRAM via fmc on STM32 MCUs?

### DISCUSSION THREAD

**Discussion by easternblack** [zephyrproject-rtos/zephyr#53737] [Q&A]
Hello.   

Recently I'm working with fmc Ethernet IC(Davicom DM9006) with STM32F7 MCUs, but there's no references.   
Of course I do not want example that works exactly with FMC ethernet, but SRAM things can be a great help for me.

Is there any FMC SRAM reference or example?

**Comment by erwango** [comment]
What about https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/drivers/memc/ram/src/main.c ?

**Reply by easternblack** [reply]
Thank you for reply.   
  
I'm work on STM32F746ZG. So I referenced stm32f746g_disco settings.   
and I also make SRAM dts node as written in dts/bindings/memory-controllers/st,stm32-fmc-nor-psram.yaml like belows.

```dts
sram {
      status = "okay";
      compatible = "st,stm32-fmc-nor-psram";

      #address-cells = <1>;
      #size-cells = <0>;

      sram2@2 {
          reg = <0x2>;
          st,control = <STM32_FMC_DATA_ADDRESS_MUX_DISABLE
                  STM32_FMC_MEMORY_TYPE_SRAM
                  STM32_FMC_NORSRAM_MEM_BUS_WIDTH_16
                  STM32_FMC_BURST_ACCESS_MODE_DISABLE
                  STM32_FMC_WAIT_SIGNAL_POLARITY_LOW
                  STM32_FMC_WAIT_TIMING_BEFORE_WS
                  STM32_FMC_WRITE_OPERATION_ENABLE
                  STM32_FMC_WAIT_SIGNAL_DISABLE
                  STM32_FMC_EXTENDED_MODE_DISABLE
                  STM32_FMC_ASYNCHRONOUS_WAIT_DISABLE
                  STM32_FMC_WRITE_BURST_DISABLE
                  STM32_FMC_CONTINUOUS_CLOCK_SYNC_ONLY
                  STM32_FMC_WRITE_FIFO_DISABLE
                  STM32_FMC_PAGE_SIZE_NONE>;
          st,timing = <3 2 4 1 0 0 STM32_FMC_ACCESS_MODE_A>;
      };
};
```
but when perform compile, DTC shows errors.
```
-- Found BOARD.dts: /home/ku/zephyrproject/zephyr/boards/arm/stm32f746g_disco/stm32f746g_disco.dts
devicetree error: /home/ku/zephyrproject/zephyr/boards/arm/stm32f746g_disco/stm32f746g_disco.dts:231 (column 19): parse error: expected number or parenthesized expression
```
line 231 means "st,control = <STM32_FMC_DATA_ADDRESS_MUX_DISABLE" on .dtsi.
    
there're some STM32 references of SDRAM when I use 'grep -rni sdram' at 'zephyr/boards/'. but there's no NOR or SRAM things on fmc. so it's very hard to know if the error above caused by my fault or another reasons.

**Reply by ramonmaruko** [reply]
You need to include `dt-bindings/memory-controller/stm32-fmc-nor-psram.h`.

**Reply by easternblack** [reply]
I found that at here. https://github.com/zephyrproject-rtos/zephyr/pull/44448#issuecomment-1100930365   
   
but thanks for your kindness
